viewer/packages/tools/src/SmartOverlay/SmartOverlayTool.ts

```diff
@@ -0,0 +1,400 @@
+/*!
+ * Copyright 2023 Cognite AS
+ */
+import { Cognite3DViewer } from '@reveal/api';
+import { assertNever, EventTrigger } from '@reveal/utilities';
+import * as THREE from 'three';
+import debounce from 'lodash/debounce';
+import { Overlay3DCollection, Overlay3DIcon } from '@reveal/3d-overlays';
+
+export type SmartOverlay = {
+  text: string;
+  id: OverlayId;
+  position: THREE.Vector3;
+  infoOverlay: HTMLElement;
+};
+
+export type OverlayInfo = {
+  text: string;
+  id: OverlayId;
+  position: THREE.Vector3;
+  color?: THREE.Color;
+};
+
+export type OverlayGroupId = string;
+export type OverlayId = number;
+
+export type OverlayGroup = {
+  id: OverlayGroupId;
+};
+
+export type OverlayToolEvent = 'hover' | 'click';
+
+export type OverlayEventHandler = (event: {
+  targetOverlay: SmartOverlay;
+  mousePosition: { clientX: number; clientY: number };
+}) => void;
+
+export type SmartOverlayToolParameters = {
+  /**
+   * Max circle size in pixels. Default is 64.
+   */
+  maxCircleSize?: number;
```

Comment:
We should point to doc on limitations in WebGL here.

Reply:
Also, you say default is 64, but below its 20. Better du have a `defaultSmartOverlayToolParameters` defined next to the type and set defaults there so they do not drift.